eternalcore-core/src/main/java/com/eternalcode/core/feature/spawn/SpawnRespawnController.java

```diff
@@ -26,8 +29,10 @@ public SpawnRespawnController(TeleportService teleportService, PluginConfigurati
     void onPlayerRespawn(PlayerRespawnEvent event) {
         Player player = event.getPlayer();
 
-        if (this.config.teleport.teleportToSpawnOnDeath) {
-            Location destinationLocation = PositionAdapter.convert(this.locations.spawn);
+        Position spawnPosition = this.locations.spawn;
+
+        if (this.config.teleport.teleportToSpawnOnDeath && !Objects.equals(spawnPosition.getWorld(), Position.NONE_WORLD)) {
+            Location destinationLocation = PositionAdapter.convert(spawnPosition);
             this.teleportService.teleport(player, destinationLocation);
```

Comment:
Co jeżeli if nie przejdzie? Gracz nie dostanie żadnej informacji o tym, co się dzieje?

Reply:
Mogę dodać ale osobiście uważam, że w tym przypadku jest to niepotrzebne to raczej oczywiste, że jeżeli ustawiasz opcję **teleportToSpawnOnDeath** to spawn musi być ustawiony